include/libsemigroups/containers.hpp

```diff
@@ -30,6 +30,88 @@
 
 namespace libsemigroups {
   namespace detail {
+    template <typename T, size_t N>
+    class StaticTriArray2 {
+      // So that StaticTriArray2<T, N> can access private data members
+      // of StaticTriArray2<S, M> and vice versa.
+      template <typename S, size_t M>
+      friend class StaticVector2;
+
```

Comment:
```suggestion
      template <typename S, size_t M>
      friend class StaticTriVector2;

```
Your probably want the firend class to have the same name as the current class, since `StaticVector2` has different private data members to your class.